README.md

```diff
@@ -301,26 +301,39 @@ Flume has support to pause/resume each pipeline.
 Once a pipeline is paused, the producer process will stop pulling jobs from the queue.
 It will process the jobs which are already pulled from the queue.
 
+Refer to "Options" section for supported options and default values.
+
+**Pause all pipelines**
+```elixir
+# Pause all pipeline permanently (in Redis) and asynchronously
+Flume.pause_all(temporary: false, async: true)
+```
+
 **Pause a pipeline**
 
 ```elixir
-# Pause the pipeline temporarily (in memory)
-Flume.pause(:default_pipeline)
+# Pause a pipeline temporarily (in memory) and asynchronously
+Flume.pause(:default_pipeline, temporary: true, async: true)
+```
 
-# Pause the pipeline permanently (in Redis)
-Flume.pause(:default_pipeline, true)
+**Resume all pipelines**
+```elixir
+# Resume all pipeline temporarily (in memory) and synchronously
+Flume.resume_all(temporary: true, async: false)
 ```
 
 **Resume a pipeline**
 
 ```elixir
-# Resume the pipeline temporarily (in memory)
-Flume.resume(:default_pipeline)
-
-# Resume the pipeline permanently (in Redis)
-Flume.resume(:default_pipeline, true)
+# Resume a pipeline permanently (in Redis) and synchronously
+Flume.resume(:default_pipeline, temporary: false, async: false)
 ```
 
+#### Options
+The following options can be used to pause/resume a pipeline
+  * `:async` - (boolean) Perform the action asynchronously. Defaults to `false`.
+  * `:temporary` - (boolean) `true` The pipeline status is updated only in-memory. `false` The pipeline status is updated in-memory and in Redis. Typically, used to maintain a pipeline control state even after an application restart. Defaults to `true`.
```

Comment:
> * `:temporary` - (boolean) `true` The pipeline status is updated only in-memory. `false` The pipeline status is updated in-memory and in Redis. Typically, used to maintain a pipeline control state even after an application restart. Defaults to `true`.

Can we re-phrase this?
- `async: true` - The caller will not wait for the operation to complete
- `async: false` - The caller will wait for the operation to complete, this can lead to timeout if the operation takes too long to succeed. See https://hexdocs.pm/elixir/GenServer.html#call/3 for more details.
- `temporary: true` - The pause/resume operation will be applied only on the current node
- `temporary: false` - Will update the value in persistent-store (Redis) and will apply the operation on all nodes

Reply:
Also for `async: false`, should we accept a timeout value from the user. 💡